cores/esp8266/i2s.h

```diff
@@ -40,17 +40,22 @@ speed.
 extern "C" {
 #endif
 
+void i2s_init();
+void i2s_deinit();
 void i2s_begin();
 void i2s_end();
 void i2s_set_rate(uint32_t rate);//Sample Rate in Hz (ex 44100, 48000)
 bool i2s_write_sample(uint32_t sample);//32bit sample with channels being upper and lower 16 bits (blocking when DMA is full)
 bool i2s_write_sample_nb(uint32_t sample);//same as above but does not block when DMA is full and returns false instead
 bool i2s_write_lr(int16_t left, int16_t right);//combines both channels and calls i2s_write_sample with the result
+bool i2s_write_lr_nb(int16_t left, int16_t right);//same as above but does not block when DMA is full and returns false instead
+uint32_t * i2s_get_buffer(); // buffer available for writing, nullptr if not applicatable.
+void i2s_put_buffer(); // yup you guessed it.
```

Comment:
sorry should provide better documentations...

the basic idea is call `i2s_get_buffer()`, and after you're done with the buffer, call `i2s_put_buffer()` to return the buffer to the subsystem.
maybe this should be called `i2s_commit_buffer()` instead.

current i2s has 8 of 64x4 bytes buffer, that's an undocumented secret to the user.
i'll improve the buffer initialization routines maybe lator.

my idea is to modify `i2s_init(size_t count, size_t size, size_t channels, size_t sizeof_sample)` to malloc() `count` of `size * channels * sizeof_sample` bytes buffers.